symless/dockerImages
# Add new OS releases

https://symless.atlassian.net/browse/S1-1676

- Adds `ubunturolling`, `fedora39`, `fedorarawhide`, and `debian12`
- Removes `ubuntu22.10`
- Refactors the testing scripts

To test:

1. Run `make rmi` to remove all existing `symless/synergy-core` docker images.
2. Run `make test-all` to test docker images.
3. Inspect the `./outputs` folder and ensure that everything is building well.
